### DISCUSSION TITLE
Is it possible to disable editing, creating or both?

### DISCUSSION THREAD

**Discussion by ralphjsmit** [filamentphp/filament#587] [Help]
Hey everyone, I love the package. Every time I browse the docs, I discover something new. Awesome!

I have the question if it is possible to restrict editing, creating or both. I already tried removing it from the `routes()` function in a resource, but the buttons were still present (and an exception that the route was not found when you clicked it).

Would be great if you could point out how to do this!

Thanks!

**Comment by wsamoht** [comment]
Thanks to @danharrin, he gave me this solution in Discord.

To disable create and delete, return `false` from `canCreate` and `canDelete` in your list class.

*Example*
```php
class ListUsers extends ListRecords
{
    public static $resource = UsersResource::class;

    public function canCreate()
    {
        return false;
    }

    public function canDelete()
    {
        return false;
    }
}
```

To disable editing, add `recordActions([])` to your table method in the resource.

*Example*
```php
// UsersResource.php
public static function table(Table $table)
{
    return $table
        ->columns([
            Text::make('name'),
        ])
        ->filters([
            //
        ])->recordActions([]);
}
```
